File: hotel-app/src/main/java/com/hotel/app/controller/HotelController.java
package com.hotel.app.controller;

import com.hotel.app.exceptions.InvalidRequestException;
import com.hotel.app.views.Hotel;
import com.hotel.app.views.BookingRequest;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;
import com.hotel.app.services.HotelService;

import java.util.List;

@Controller
@RestController
@RequestMapping("/api")

public class HotelController {
    private final HotelService hotelService;

    public HotelController(HotelService hotelService) {
        this.hotelService = hotelService;
    }

    @GetMapping("/search/hotels")
    public List<Hotel> searchHotels(@RequestParam String city) {
        return hotelService.searchHotels(city);
    }

    @PostMapping("/bookings")
    public ResponseEntity<String> BookHotel(@RequestBody BookingRequest bookingRequest,   @AuthenticationPrincipal UserDetails userDetails) throws InvalidRequestException {
        try {
            String userId = userDetails.getUsername();
            hotelService.book(bookingRequest,userId);
            return ResponseEntity.ok("message:booking successful");
        } catch (InvalidRequestException e) {
            return ResponseEntity.badRequest().body(e.getMessage());
        }
    }

}

File: hotel-service/hotel_controller.ts
// @ts-ignore
// checking push
import { Hono } from "hono";
import { logger } from "hono/logger";
// @ts-ignore
import * as hotelBooking from "./hotel_service.ts";
const app = new Hono();
console.log("running...");
app.use(logger());

app.post("/api/bookings", async (c) => {
  const token = c.req.header("Authorization");
  const bookingRequest = await c.req.json();
  await hotelBooking.book(bookingRequest, token);
  return c.text("Successful booking");
});

app.get("/api/search/hotels", async (c) => {
  const city = c.req.query("city");
  const result = await hotelBooking.searchHotels(city);
  return c.json(result);
});

app.get("/", (c) => c.json("hello"));
export default app;
